tests/integration/conftest.py

```diff
@@ -33,20 +36,56 @@ def traefik_app_name_fixture() -> str:
     return "traefik-k8s"
 
 
+@pytest_asyncio.fixture(scope="module", name="build_charm")
+async def build_charm_fixture(ops_test) -> str:
+    """Builds the charm and injects additional configurations into config.yaml.
+
+    This fixture is designed to simulate a feature that is not yet available in charmcraft that
+    allows for the modification of charm configurations during the build process.
+    Three additional configurations, namely flask_foo_str, flask_foo_int, and flask_foo_bool,
+    will be appended to the config.yaml file.
+    """
+    charm = await ops_test.build_charm(".")
+    charm_zip = zipfile.ZipFile(charm, "r")
+    with charm_zip.open("config.yaml") as file:
+        config = yaml.safe_load(file)
+    config["options"].update(
+        {
+            "flask_foo_str": {"type": "string"},
+            "flask_foo_int": {"type": "int"},
+            "flask_foo_bool": {"type": "boolean"},
+        }
+    )
+    modified_config = yaml.safe_dump(config)
+    new_charm = io.BytesIO()
+    with zipfile.ZipFile(new_charm, "w") as new_charm_zip:
+        for item in charm_zip.infolist():
+            if item.filename == "config.yaml":
+                new_charm_zip.writestr(item, modified_config)
+            else:
+                with charm_zip.open(item) as file:
+                    data = file.read()
+                new_charm_zip.writestr(item, data)
+    charm_zip.close()
```

Comment:
Isn't there a way to replace only `config.yaml` instead of recreating a new zip ?

Reply:
Python `zipfile` library doesn't support modify zip file in place.